### DISCUSSION TITLE
How to get access to appender methods with SLF4J?

### DISCUSSION THREAD

**Discussion by emelyanovkr** [apache/logging-log4j2#2572] [Q&A] [answered]
I am getting my Logger like this `private final Logger LOGGER = LoggerFactory.getLogger("TestLogger")`. I am using `ListAppender `behind `TestLogger` name, so I want to get access to `ListAppender` fields and get the `List of LogEvents` to write unit tests ensuring messages are capturing as needed,

I am using `log4j-core-test 2.23.1` and `slf4j 2.0.12`.

**Comment by ppkarwasz** [comment, marked as answer]
You just need to use `ListAppender.getListAppender()` with the name of your appender:

https://github.com/apache/logging-log4j2/blob/fe92961737ed3423763df6219e892b2d1a3ad49b/log4j-core-test/src/main/java/org/apache/logging/log4j/core/test/appender/ListAppender.java#L292-L302

**Reply by emelyanovkr** [reply]
So that's simple, I was going throguh the code but seems like I wasn't checking carefully, so I missed out this method. Thank you!
